experiments/backtest.py

```diff
@@ -166,11 +180,35 @@ def interest_and_fees(
     return cash_interest + shorting_fee
 
 
+@dataclass
+class Timing:
+    solver: float
+    cvxpy: float
+    other: float
+
+    @property
+    def total(self):
+        return self.solver + self.cvxpy + self.other
+
+    @classmethod
+    def get_timing(
+        cls, start_time: float, end_time: float, problem: cp.Problem | None
+    ) -> Timing:
+        if problem:
+            solver_time = problem.solver_stats.solve_time
+            cvxpy_time = problem.compilation_time
+            other_time = end_time - start_time - solver_time - cvxpy_time
+            return cls(solver_time, cvxpy_time, other_time)
+        else:
+            return cls(0, 0, 0)
```

Comment:
@kasperjo @tschm I've added a timer so we can create the stacked area chart for the backtest. It required returning the problem from the `markowitz` function. I don't think this is a problem, though.